Mathlib/GroupTheory/SpecificGroups/Dihedral.lean

```diff
@@ -90,6 +90,18 @@ theorem sr_mul_r (i j : ZMod n) : sr i * r j = sr (i + j) :=
 theorem sr_mul_sr (i j : ZMod n) : sr i * sr j = r (j - i) :=
   rfl
 
+@[simp]
+theorem inv_r (i : ZMod n) : (r i)⁻¹ = r (-i) := by
+  rfl
+
+@[simp]
+theorem inv_sr (i : ZMod n) : (sr i)⁻¹ = sr (i) := by
+  rfl
+
+@[simp]
+theorem r_zero (i : ZMod n) : (r i) ^ 0 = r 0 := by
```

Comment:
Sorry, I meant `r 0 = 1`.

Reply:
Should `r 0` really simplify to `1`? I had a proof that looks something like this where I want to rewrite it the other way

```
example (h : (r (i : ZMod n)) = 1) : i = 0 := by
  have : 1 = r (0 : ZMod n) := by rfl
  rw [this] at h
  simp at h
  exact h
```

Reply:
I also realised that the equality `r 0 = 1` is already stated as `one_def`

Reply:
I'll also delete the equality `(r i) ^ 0 = r 0` as I've never needed it

Reply:
Of course you could and might want to rewrite `1` with `r 0` in proofs, but `1` is simpler than `r 0` so the simplifier should still fire on it.

Reply:
My point is more that I was using the `simp` tactic as a "normal form rewriting" tactic (rewrite all terms of type `DihedralGroup` to either `r <something>` or `sr <something>`), but I do see that normal forms can be quite complicated in general and it might not be a good idea to use `simp` for this purpose

Reply:
Adding `r 0 = 1` as a simp theorem seems to mess up the proof in `OddCommuteEquiv` in a way that I don't know how to fix

Reply:
You are right that `simp` is a "normal form rewriting" tactic. We call the result the "simp normal form". What I am arguing is that the simp normal form of an element should be `r <some non-zero thing>` or `sr <something>` or `1`. This is because there are plenty lemmas applicable to `1` in general groups, but `r 0` is specific to dihedral groups so less lemmas could fire on it.

But of course this is not the main focus of this PR.